### ISSUE TITLE
Can't compile for iOS 

### ISSUE THREAD

**Issue by codeservice** [HaxeFoundation/hxcpp#937]
Error message:

Error: /usr/local/lib/haxe/lib/hxcpp/4,0,0/src/hx/Thread.cpp:341:8: error: thread-local storage is not supported for the current target
static thread_local ThreadInfoHolder threadHolder;

**Comment by hughsando** [comment]
Looks like this might be a problem compiling for i386 (simulator) or maybe iOS<version 9.  Is this the case?
The fix is the change the condition `__cplusplus > 199711L` in Thread.cpp to exclude the case where it does not work - just not 100% sure when this is at the moment.

**Comment by codeservice** [comment]
Yes, I was compiling for simulator on latest xCode. Probably possible exclude this case from Thread.

**Comment by hughsando** [comment]
Was it 32 bit simulator? (armv7 for i386, not x86_64 for arm64).  The x86 path is becoming less relevant - not sure it is worth supporting.  Maybe the default simulator is not 64-bit?  That would be sub-optimal.

**Comment by codeservice** [comment]
Compile error rise when I was compiling my project on this step:
haxelib run hxcpp Build.xml -Diphonesim -DHXCPP_M64 -DHXCPP_CPP11 -debug

Compiling group: runtime
xcrun --sdk iphonesimulator14.3 clang++ -D_CRT_SECURE_NO_DEPRECATE -DHX_UNDEFINE_H -c -g -fmessage-length=0 -pipe -Wno-trigraphs -fno-stack-protector -Wno-overflow -fpascal-strings -fasm-blocks -Wreturn-type -Wno-unused-variable -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator14.3.sdk -miphoneos-version-min=5.0 -DIPHONE=IPHONE -DIPHONESIM=IPHONESIM -Wno-parentheses -Wno-null-dereference -Wno-unused-value -Wno-bool-conversion -DHXCPP_DEBUG -DSTATIC_LINK(static) -DHXCPP_M64 -DHXCPP_VISIT_ALLOCS(haxe) -DHXCPP_CHECK_POINTER(haxe) -DHXCPP_API_LEVEL=331(haxe) -DHXCPP_OBJC(haxe) -I/usr/local/lib/haxe/lib/hxcpp/4,0,0/include -fexceptions -fstrict-aliasing ... tags=[haxe]
 - src/hx/Thread.cpp 
 - src/hx/gc/Immix.cpp  [haxe,gc]
Error: /usr/local/lib/haxe/lib/hxcpp/4,0,0/src/hx/Thread.cpp:341:8: error: thread-local storage is not supported for the current target
static thread_local ThreadInfoHolder threadHolder;
       ^
1 error generated.


Does look like:  -arch x86_64

**Comment by codeservice** [comment]
Hmm, it works if I change this -miphoneos-version-min=5.0  to -miphoneos-version-min=9.0 in toolchain for iphonesim.

**Comment by justin-espedal** [comment]
Related (to the last comment, not this thread in general): https://github.com/haxelime/lime/issues/1429#issuecomment-721531498

Is there a reason that value is hardcoded there? If so, perhaps it's about time to increase it. Though I don't know what a more sensible number would be.

**Comment by codeservice** [comment]
I can confirm -miphoneos-version-min=6.0 works too

**Comment by codeservice** [comment]
Oops, sorry. -miphoneos-version-min=6.0  Doesnt work. Still the same issue. I will try 7,8. Know for sure 9.0 works

**Comment by codeservice** [comment]
So 5,6,7 doesn't work. Real minimum version for fixing this issue: -miphoneos-version-min=8.0

**Comment by codeservice** [comment]
FYI
For armv7 min version 8.0 doesn't work. Required -miphoneos-version-min=9.0 
Weird, no consistency whatsoever. Don't understand why 8.0 enough for simulator.

**Comment by hughsando** [comment]
Yes, I've gone with v9 (2016).  Should be fine I think.

**Comment by andrT** [comment]
Not sure if anything has changed since this was closed, but I am getting the same errors with hxcpp 4.2.1. In the logs I see that indeed -miphoneos-version-min=9.0 is set, but it is still not working.

Here's a very simple example:
```
#include <thread>
thread_local int threadID;
int main() {
    return 0;
}
```
`xcrun --sdk iphonesimulator16.2 clang++ -v -arch i386 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator16.2.sdk -miphoneos-version-min=9.0 -x c++ -stdlib=libc++ -std=c++11 test.cpp -otest.o`

I receive the same error. However, this works once I set -miphoneos-version-min=10.0.

This error is essentially created in llvm-project/clang/lib/Basic/Targets/OSTargets.h in DarwinTargetInfo function.
if (Triple.isMacOSX())
  this->TLSSupported = !Triple.isMacOSXVersionLT(10, 7);
else if (Triple.isiOS()) {
  // 64-bit iOS supported it from 8 onwards, 32-bit device from 9 onwards,
  // 32-bit simulator from 10 onwards.
  if (Triple.isArch64Bit())
    this->TLSSupported = !Triple.isOSVersionLT(8);
  else if (Triple.isArch32Bit()) {
    if (!Triple.isSimulatorEnvironment())
      this->TLSSupported = !Triple.isOSVersionLT(9);
    else
      this->TLSSupported = !Triple.isOSVersionLT(10);
  }
}

This current state of this code was done by 4ea49798a9f8a on Jan 5 2018. How about bumping the version from 9 to 10?
